zk/groth16/src/proof/mod.rs

```diff
@@ -53,6 +56,46 @@ impl<E: CompressSize> CompressedProof<E> {
     }
 }
 
+impl CompressedProof<Bn254> {
+
+/// Total size = G1 + G2 + G1 at the type level.
+    pub fn to_bytes(
+        &self,
+    ) -> [u8; 128]
+    {
+        let mut bytes = [0u8; 128];
+        let g1 = <Bn254 as CompressSize>::G1CompressedSize::to_usize();
+        let g2 = <Bn254 as CompressSize>::G2CompressedSize::to_usize();
+
+        bytes[..g1].copy_from_slice(&self.pi_a);
+        bytes[g1..g1 + g2].copy_from_slice(&self.pi_b);
+        bytes[g1 + g2..].copy_from_slice(&self.pi_c);
+
+        bytes
+    }
+
+    /// Type-level length bound: accepts exactly G1 + G2 + G1 bytes.
+    pub fn from_bytes(bytes: &[u8; 128]) -> Self {
+        let g1 = <Bn254 as CompressSize>::G1CompressedSize::to_usize();
+        let g2 = <Bn254 as CompressSize>::G2CompressedSize::to_usize();
+
+        let mut pi_a = GenericArray::default();
+        let mut pi_b = GenericArray::default();
+        let mut pi_c = GenericArray::default();
+
+        pi_a.copy_from_slice(&bytes[..g1]);
+        pi_b.copy_from_slice(&bytes[g1..g1 + g2]);
+        pi_c.copy_from_slice(&bytes[g1 + g2..]);
+
+        Self {
+            pi_a,
+            pi_b,
+            pi_c,
+            _pairing: PhantomData,
+        }
+    }
+}
```

Comment:
Can it be a generic implementation? This is fine as it is all we need. But if we could leverage the info so its not direcly implemented for `Bn254` would be better. Not sure if its possible tho.